From 52e1ca88d13dcb57d8a54e21eaf4cbde3faab535 Mon Sep 17 00:00:00 2001
From: Iceee <andrew@opticgaming.tv>
Date: Tue, 4 Aug 2015 18:15:05 -0700
Subject: [PATCH] Fix lava/water some times creating air instead of cobblestone


diff --git a/src/main/java/net/minecraft/server/BlockFluids.java b/src/main/java/net/minecraft/server/BlockFluids.java
index b6104502..db73f5d9 100644
--- a/src/main/java/net/minecraft/server/BlockFluids.java
+++ b/src/main/java/net/minecraft/server/BlockFluids.java
@@ -154,7 +154,7 @@ public abstract class BlockFluids extends Block {
                     return true;
                 }
 
-                if (integer.intValue() <= 4) {
+                if (integer.intValue() > 0) { // PaperSpigot
                     world.setTypeUpdate(blockposition, Blocks.COBBLESTONE.getBlockData());
                     this.fizz(world, blockposition);
                     return true;
-- 
2.32.0

