From 6aa4f968c399d88036dc080458a904498c1c9e63 Mon Sep 17 00:00:00 2001
From: LinsaFTW <25271111+linsaftw@users.noreply.github.com>
Date: Thu, 27 Oct 2022 12:56:16 +0200
Subject: [PATCH] Disable deprecation mappings


diff --git a/deprecation-mappings.at b/deprecation-mappings.at
deleted file mode 100644
index 5858b4887..000000000
--- a/deprecation-mappings.at
+++ /dev/null
@@ -1,62 +0,0 @@
-public+synthetic org/bukkit/Bukkit/getOnlinePlayers()[Lorg/bukkit/entity/Player;
-public+synthetic org/bukkit/Server/getOnlinePlayers()[Lorg/bukkit/entity/Player;
-
-public+synthetic org/bukkit/entity/Damageable/damage(I)V
-public+synthetic org/bukkit/entity/Damageable/damage(ILorg/bukkit/entity/Entity;)V
-public+synthetic org/bukkit/entity/Damageable/getHealth()I
-public+synthetic org/bukkit/entity/Damageable/getMaxHealth()I
-public+synthetic org/bukkit/entity/Damageable/setHealth(I)V
-public+synthetic org/bukkit/entity/Damageable/setMaxHealth(I)V
-
-public+synthetic org/bukkit/entity/LivingEntity/getLastDamage()I
-public+synthetic org/bukkit/entity/LivingEntity/setLastDamage(I)V
-
-public+synthetic org/bukkit/entity/Minecart/getDamage()I
-public+synthetic org/bukkit/entity/Minecart/setDamage(I)V
-
-public+synthetic org/bukkit/entity/Projectile/getShooter()Lorg/bukkit/entity/LivingEntity;
-public+synthetic org/bukkit/entity/Projectile/setShooter(Lorg/bukkit/entity/LivingEntity;)V
-
-public+synthetic org/bukkit/event/entity/EntityDamageEvent/getDamage()I
-public+synthetic org/bukkit/event/entity/EntityDamageEvent/setDamage(I)V
-
-public+synthetic org/bukkit/event/entity/EntityRegainHealthEvent/getAmount()I
-public+synthetic org/bukkit/event/entity/EntityRegainHealthEvent/setAmount(I)V
-
-public+synthetic org/bukkit/event/vehicle/VehicleDamageEvent/getDamage()I
-public+synthetic org/bukkit/event/vehicle/VehicleDamageEvent/setDamage(I)V
-
-# CraftBukkit
-public+synthetic org/bukkit/craftbukkit/v1_8_R3/CraftServer/getOnlinePlayers()[Lorg/bukkit/entity/Player;
-
-public+synthetic org/bukkit/craftbukkit/v1_8_R3/entity/CraftArrow/getShooter()Lorg/bukkit/entity/LivingEntity;
-public+synthetic org/bukkit/craftbukkit/v1_8_R3/entity/CraftArrow/setShooter(Lorg/bukkit/entity/LivingEntity;)V
-
-public+synthetic org/bukkit/craftbukkit/v1_8_R3/entity/CraftEnderDragonPart/damage(I)V
-public+synthetic org/bukkit/craftbukkit/v1_8_R3/entity/CraftEnderDragonPart/damage(ILorg/bukkit/entity/Entity;)V
-public+synthetic org/bukkit/craftbukkit/v1_8_R3/entity/CraftEnderDragonPart/getHealth()I
-public+synthetic org/bukkit/craftbukkit/v1_8_R3/entity/CraftEnderDragonPart/getMaxHealth()I
-public+synthetic org/bukkit/craftbukkit/v1_8_R3/entity/CraftEnderDragonPart/setHealth(I)V
-public+synthetic org/bukkit/craftbukkit/v1_8_R3/entity/CraftEnderDragonPart/setMaxHealth(I)V
-
-
-public+synthetic org/bukkit/craftbukkit/v1_8_R3/entity/CraftFireball/getShooter()Lorg/bukkit/entity/LivingEntity;
-public+synthetic org/bukkit/craftbukkit/v1_8_R3/entity/CraftFireball/setShooter(Lorg/bukkit/entity/LivingEntity;)V
-
-public+synthetic org/bukkit/craftbukkit/v1_8_R3/entity/CraftFish/getShooter()Lorg/bukkit/entity/LivingEntity;
-public+synthetic org/bukkit/craftbukkit/v1_8_R3/entity/CraftFish/setShooter(Lorg/bukkit/entity/LivingEntity;)V
-
-public+synthetic org/bukkit/craftbukkit/v1_8_R3/entity/CraftLivingEntity/damage(I)V
-public+synthetic org/bukkit/craftbukkit/v1_8_R3/entity/CraftLivingEntity/damage(ILorg/bukkit/entity/Entity;)V
-public+synthetic org/bukkit/craftbukkit/v1_8_R3/entity/CraftLivingEntity/getHealth()I
-public+synthetic org/bukkit/craftbukkit/v1_8_R3/entity/CraftLivingEntity/getMaxHealth()I
-public+synthetic org/bukkit/craftbukkit/v1_8_R3/entity/CraftLivingEntity/setHealth(I)V
-public+synthetic org/bukkit/craftbukkit/v1_8_R3/entity/CraftLivingEntity/setMaxHealth(I)V
-public+synthetic org/bukkit/craftbukkit/v1_8_R3/entity/CraftLivingEntity/getLastDamage()I
-public+synthetic org/bukkit/craftbukkit/v1_8_R3/entity/CraftLivingEntity/setLastDamage(I)V
-
-public+synthetic org/bukkit/craftbukkit/v1_8_R3/entity/CraftMinecart/getDamage()I
-public+synthetic org/bukkit/craftbukkit/v1_8_R3/entity/CraftMinecart/setDamage(I)V
-
-public+synthetic org/bukkit/craftbukkit/v1_8_R3/entity/CraftProjectile/getShooter()Lorg/bukkit/entity/LivingEntity;
-public+synthetic org/bukkit/craftbukkit/v1_8_R3/entity/CraftProjectile/setShooter(Lorg/bukkit/entity/LivingEntity;)V
diff --git a/deprecation-mappings.csrg b/deprecation-mappings.csrg
deleted file mode 100644
index 48bc50495..000000000
--- a/deprecation-mappings.csrg
+++ /dev/null
@@ -1,27 +0,0 @@
-org/bukkit/Bukkit _INVALID_getOnlinePlayers ()[Lorg/bukkit/entity/Player; getOnlinePlayers
-org/bukkit/Server _INVALID_getOnlinePlayers ()[Lorg/bukkit/entity/Player; getOnlinePlayers
-
-org/bukkit/entity/Damageable _INVALID_damage (I)V damage
-org/bukkit/entity/Damageable _INVALID_damage (ILorg/bukkit/entity/Entity;)V damage
-org/bukkit/entity/Damageable _INVALID_getHealth ()I getHealth
-org/bukkit/entity/Damageable _INVALID_getMaxHealth ()I getMaxHealth
-org/bukkit/entity/Damageable _INVALID_setHealth (I)V setHealth
-org/bukkit/entity/Damageable _INVALID_setMaxHealth (I)V setMaxHealth
-
-org/bukkit/entity/LivingEntity _INVALID_getLastDamage ()I getLastDamage
-org/bukkit/entity/LivingEntity _INVALID_setLastDamage (I)V setLastDamage
-
-org/bukkit/entity/Minecart _INVALID_getDamage ()I getDamage
-org/bukkit/entity/Minecart _INVALID_setDamage (I)V setDamage
-
-org/bukkit/entity/Projectile _INVALID_getShooter ()Lorg/bukkit/entity/LivingEntity; getShooter
-org/bukkit/entity/Projectile _INVALID_setShooter (Lorg/bukkit/entity/LivingEntity;)V setShooter
-
-org/bukkit/event/entity/EntityDamageEvent _INVALID_getDamage ()I getDamage
-org/bukkit/event/entity/EntityDamageEvent _INVALID_setDamage (I)V setDamage
-
-org/bukkit/event/entity/EntityRegainHealthEvent _INVALID_getAmount ()I getAmount
-org/bukkit/event/entity/EntityRegainHealthEvent _INVALID_setAmount (I)V setAmount
-
-org/bukkit/event/vehicle/VehicleDamageEvent _INVALID_getDamage ()I getDamage
-org/bukkit/event/vehicle/VehicleDamageEvent _INVALID_setDamage (I)V setDamage
diff --git a/pom.xml b/pom.xml
index 421dd106e..98c82109b 100644
--- a/pom.xml
+++ b/pom.xml
@@ -226,24 +226,6 @@
                     </execution>
                 </executions>
             </plugin>
-            <!-- Gotta go after shade plugin -->
-            <plugin>
-                <groupId>net.md-5</groupId>
-                <artifactId>specialsource-maven-plugin</artifactId>
-		<version>1.2.3</version> <!-- paper - bump log4j -->
-                <executions>
-                    <execution>
-                        <phase>package</phase>
-                        <goals>
-                            <goal>remap</goal>
-                        </goals>
-                        <configuration>
-                            <srgIn>${project.basedir}/deprecation-mappings.csrg</srgIn>
-                            <accessIn>${project.basedir}/deprecation-mappings.at</accessIn>
-                        </configuration>
-                    </execution>
-                </executions>
-            </plugin>
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-compiler-plugin</artifactId>
-- 
2.37.3.windows.1

