From 0bb7ec526e4a11db974d9107ffa45ca93a7f0128 Mon Sep 17 00:00:00 2001
From: LinsaFTW <25271111+linsaftw@users.noreply.github.com>
Date: Fri, 13 Jan 2023 17:25:26 -0300
Subject: [PATCH] Update Deprecation Mappings


diff --git a/deprecation-mappings.csrg b/deprecation-mappings.csrg
index 48bc50495..eafeaf211 100644
--- a/deprecation-mappings.csrg
+++ b/deprecation-mappings.csrg
@@ -1,27 +1,20 @@
 org/bukkit/Bukkit _INVALID_getOnlinePlayers ()[Lorg/bukkit/entity/Player; getOnlinePlayers
 org/bukkit/Server _INVALID_getOnlinePlayers ()[Lorg/bukkit/entity/Player; getOnlinePlayers
-
 org/bukkit/entity/Damageable _INVALID_damage (I)V damage
 org/bukkit/entity/Damageable _INVALID_damage (ILorg/bukkit/entity/Entity;)V damage
 org/bukkit/entity/Damageable _INVALID_getHealth ()I getHealth
 org/bukkit/entity/Damageable _INVALID_getMaxHealth ()I getMaxHealth
 org/bukkit/entity/Damageable _INVALID_setHealth (I)V setHealth
 org/bukkit/entity/Damageable _INVALID_setMaxHealth (I)V setMaxHealth
-
 org/bukkit/entity/LivingEntity _INVALID_getLastDamage ()I getLastDamage
 org/bukkit/entity/LivingEntity _INVALID_setLastDamage (I)V setLastDamage
-
 org/bukkit/entity/Minecart _INVALID_getDamage ()I getDamage
 org/bukkit/entity/Minecart _INVALID_setDamage (I)V setDamage
-
 org/bukkit/entity/Projectile _INVALID_getShooter ()Lorg/bukkit/entity/LivingEntity; getShooter
 org/bukkit/entity/Projectile _INVALID_setShooter (Lorg/bukkit/entity/LivingEntity;)V setShooter
-
 org/bukkit/event/entity/EntityDamageEvent _INVALID_getDamage ()I getDamage
 org/bukkit/event/entity/EntityDamageEvent _INVALID_setDamage (I)V setDamage
-
 org/bukkit/event/entity/EntityRegainHealthEvent _INVALID_getAmount ()I getAmount
 org/bukkit/event/entity/EntityRegainHealthEvent _INVALID_setAmount (I)V setAmount
-
 org/bukkit/event/vehicle/VehicleDamageEvent _INVALID_getDamage ()I getDamage
-org/bukkit/event/vehicle/VehicleDamageEvent _INVALID_setDamage (I)V setDamage
+org/bukkit/event/vehicle/VehicleDamageEvent _INVALID_setDamage (I)V setDamage
\ No newline at end of file
diff --git a/pom.xml b/pom.xml
index 232cfcfe0..acd2e3e6a 100644
--- a/pom.xml
+++ b/pom.xml
@@ -241,7 +241,7 @@
             <plugin>
                 <groupId>net.md-5</groupId>
                 <artifactId>specialsource-maven-plugin</artifactId>
-		<version>1.2.3</version> <!-- paper - bump log4j -->
+		            <version>1.2.4</version> <!-- FlamePaper - Update Deprecation Mappings-->
                 <executions>
                     <execution>
                         <phase>package</phase>
@@ -250,7 +250,6 @@
                         </goals>
                         <configuration>
                             <srgIn>${project.basedir}/deprecation-mappings.csrg</srgIn>
-                            <accessIn>${project.basedir}/deprecation-mappings.at</accessIn>
                         </configuration>
                     </execution>
                 </executions>
-- 
2.37.3.windows.1

