From e487d88c94bf1e9eb1bb56eff4b7ae94f2c3198a Mon Sep 17 00:00:00 2001
From: LinsaFTW <25271111+linsaftw@users.noreply.github.com>
Date: Fri, 13 Jan 2023 17:25:26 -0300
Subject: [PATCH] Update Deprecation Mappings


diff --git a/deprecation-mappings.csrg b/deprecation-mappings.csrg
index 48bc50495..b8e4a5dd7 100644
--- a/deprecation-mappings.csrg
+++ b/deprecation-mappings.csrg
@@ -1,27 +1,27 @@
-org/bukkit/Bukkit _INVALID_getOnlinePlayers ()[Lorg/bukkit/entity/Player; getOnlinePlayers
-org/bukkit/Server _INVALID_getOnlinePlayers ()[Lorg/bukkit/entity/Player; getOnlinePlayers
+org/bukkit/Bukkit getOnlinePlayers ()[Lorg/bukkit/entity/Player; _INVALID_getOnlinePlayers
+org/bukkit/Server getOnlinePlayers ()[Lorg/bukkit/entity/Player; _INVALID_getOnlinePlayers
 
-org/bukkit/entity/Damageable _INVALID_damage (I)V damage
-org/bukkit/entity/Damageable _INVALID_damage (ILorg/bukkit/entity/Entity;)V damage
-org/bukkit/entity/Damageable _INVALID_getHealth ()I getHealth
-org/bukkit/entity/Damageable _INVALID_getMaxHealth ()I getMaxHealth
-org/bukkit/entity/Damageable _INVALID_setHealth (I)V setHealth
-org/bukkit/entity/Damageable _INVALID_setMaxHealth (I)V setMaxHealth
+org/bukkit/entity/Damageable damage (I)V _INVALID_damage
+org/bukkit/entity/Damageable damage (ILorg/bukkit/entity/Entity;)V _INVALID_damage
+org/bukkit/entity/Damageable getHealth ()I _INVALID_getHealth
+org/bukkit/entity/Damageable getMaxHealth ()I _INVALID_getMaxHealth
+org/bukkit/entity/Damageable setHealth (I)V _INVALID_setHealth
+org/bukkit/entity/Damageable setMaxHealth (I)V _INVALID_setMaxHealth
 
-org/bukkit/entity/LivingEntity _INVALID_getLastDamage ()I getLastDamage
-org/bukkit/entity/LivingEntity _INVALID_setLastDamage (I)V setLastDamage
+org/bukkit/entity/LivingEntity getLastDamage ()I _INVALID_getLastDamage
+org/bukkit/entity/LivingEntity setLastDamage (I)V _INVALID_setLastDamage
 
-org/bukkit/entity/Minecart _INVALID_getDamage ()I getDamage
-org/bukkit/entity/Minecart _INVALID_setDamage (I)V setDamage
+org/bukkit/entity/Minecart getDamage ()I _INVALID_getDamage
+org/bukkit/entity/Minecart setDamage (I)V _INVALID_setDamage
 
-org/bukkit/entity/Projectile _INVALID_getShooter ()Lorg/bukkit/entity/LivingEntity; getShooter
-org/bukkit/entity/Projectile _INVALID_setShooter (Lorg/bukkit/entity/LivingEntity;)V setShooter
+org/bukkit/entity/Projectile getShooter ()Lorg/bukkit/entity/LivingEntity; _INVALID_getShooter
+org/bukkit/entity/Projectile setShooter (Lorg/bukkit/entity/LivingEntity;)V _INVALID_setShooter
 
-org/bukkit/event/entity/EntityDamageEvent _INVALID_getDamage ()I getDamage
-org/bukkit/event/entity/EntityDamageEvent _INVALID_setDamage (I)V setDamage
+org/bukkit/event/entity/EntityDamageEvent getDamage ()I _INVALID_getDamage
+org/bukkit/event/entity/EntityDamageEvent setDamage (I)V _INVALID_setDamage
 
-org/bukkit/event/entity/EntityRegainHealthEvent _INVALID_getAmount ()I getAmount
-org/bukkit/event/entity/EntityRegainHealthEvent _INVALID_setAmount (I)V setAmount
+org/bukkit/event/entity/EntityRegainHealthEvent getAmount ()I _INVALID_getAmount
+org/bukkit/event/entity/EntityRegainHealthEvent setAmount (I)V _INVALID_setAmount
 
-org/bukkit/event/vehicle/VehicleDamageEvent _INVALID_getDamage ()I getDamage
-org/bukkit/event/vehicle/VehicleDamageEvent _INVALID_setDamage (I)V setDamage
+org/bukkit/event/vehicle/VehicleDamageEvent getDamage ()I _INVALID_getDamage
+org/bukkit/event/vehicle/VehicleDamageEvent setDamage (I)V _INVALID_setDamage
\ No newline at end of file
diff --git a/pom.xml b/pom.xml
index f42979759..6a9bf72cb 100644
--- a/pom.xml
+++ b/pom.xml
@@ -236,7 +236,7 @@
             <plugin>
                 <groupId>net.md-5</groupId>
                 <artifactId>specialsource-maven-plugin</artifactId>
-		<version>1.2.3</version> <!-- paper - bump log4j -->
+		            <version>1.2.4</version> <!-- FlamePaper - Update Deprecation Mappings-->
                 <executions>
                     <execution>
                         <phase>package</phase>
-- 
2.37.3.windows.1

