From 7fc625812e5d1826ffd43dff7f4dad6ef38e046e Mon Sep 17 00:00:00 2001
From: LinsaFTW <25271111+linsaftw@users.noreply.github.com>
Date: Fri, 27 Jan 2023 11:00:47 -0300
Subject: [PATCH] Dont Handle Overflow Inventory Clicks


diff --git a/src/main/java/net/minecraft/server/Container.java b/src/main/java/net/minecraft/server/Container.java
index 943f068fc..62c66b4e0 100644
--- a/src/main/java/net/minecraft/server/Container.java
+++ b/src/main/java/net/minecraft/server/Container.java
@@ -249,24 +249,7 @@ public abstract class Container {
                             }
                         }
                     }
-                } else if (k == 1) {
-                    if (i < 0) {
-                        return null;
-                    }
-
-                    slot2 = (Slot) this.c.get(i);
-                    if (slot2 != null && slot2.isAllowed(entityhuman)) {
-                        itemstack1 = this.b(entityhuman, i);
-                        if (itemstack1 != null) {
-                            Item item = itemstack1.getItem();
-
-                            itemstack = itemstack1.cloneItemStack();
-                            if (slot2.getItem() != null && slot2.getItem().getItem() == item) {
-                                this.a(i, j, true, entityhuman);
-                            }
-                        }
-                    }
-                } else {
+                } else { // FlamePaper - Dont Handle Overflow Inventory Clicks
                     if (i < 0) {
                         return null;
                     }
-- 
2.37.3.windows.1

